src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -568,9 +593,18 @@ public virtual string NormalizeKey(string key)
             var result = await VerifyPasswordAsync(passwordStore, user, password, cancellationToken);
             if (result == PasswordVerificationResult.SuccessRehashNeeded) {
                 await UpdatePasswordHash(passwordStore, user, password, cancellationToken, validatePassword: false);
-                await UpdateAsync(user, cancellationToken);
+                await UpdateUserAsync(user, cancellationToken);
+            }
+            if (result != PasswordVerificationResult.Failed)
+            {
+                await LogResultAsync(IdentityResult.Success, user);
+            }
+            else
+            {
+                await LogResultAsync(IdentityResult.Failed(ErrorDescriber.IncorrectPassword()), user);
```

Comment:
I think you should have an overload of LogResult that takes a bool and just prints out, instead of introducing a new resource that isn't used except for logging

CheckPasswordAsync for user: whatever : True | False